segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -412,33 +415,32 @@ public static void updateCoreAttributes(Map<Integer, String> backUpMetadataSegme
+    public static synchronized CompletableFuture<Map<Integer, String>> createBackUpMetadataSegments(Storage storage, int containerCount, ExecutorService executorService,
+                                                                    Duration timeout) {
         String fileSuffix = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
         Map<Integer, String> backUpMetadataSegments = new HashMap<>();
 
         val futures = new ArrayList<CompletableFuture<Void>>();
 
-        for (int containerId = 0; containerId < containerCount; containerId++) {
-            String backUpMetadataSegment = NameUtils.getMetadataSegmentName(containerId) + fileSuffix;
-            String backUpAttributeSegment = NameUtils.getAttributeSegmentName(backUpMetadataSegment);
-            log.debug("Created '{}' as a back of metadata segment of container Id '{}'", backUpMetadataSegment, containerId);
-
-            val finalContainerId = containerId;
-            futures.add(Futures.exceptionallyExpecting(
-                    ContainerRecoveryUtils.backUpMetadataAndAttributeSegments(storage, containerId,
-                            backUpMetadataSegment, backUpAttributeSegment, executorService, timeout)
-                            .thenAccept(x -> ContainerRecoveryUtils.deleteMetadataAndAttributeSegments(storage, finalContainerId, timeout)
-                                    .thenAccept(z -> backUpMetadataSegments.put(finalContainerId, backUpMetadataSegment))
-                            ), ex -> Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException, null));
-        }
-        Futures.allOf(futures).get(timeout.toMillis(), TimeUnit.MILLISECONDS);
-        return backUpMetadataSegments;
+        return CompletableFuture.supplyAsync(() -> {
+            for (int containerId = 0; containerId < containerCount; containerId++) {
+                String backUpMetadataSegment = NameUtils.getMetadataSegmentName(containerId) + fileSuffix;
+                String backUpAttributeSegment = NameUtils.getAttributeSegmentName(backUpMetadataSegment);
+                log.debug("Created '{}' as a back of metadata segment of container Id '{}'", backUpMetadataSegment, containerId);
+
+                val finalContainerId = containerId;
+                futures.add(Futures.exceptionallyExpecting(
+                        ContainerRecoveryUtils.backUpMetadataAndAttributeSegments(storage, containerId,
+                                backUpMetadataSegment, backUpAttributeSegment, executorService, timeout)
+                                .thenCompose(x -> ContainerRecoveryUtils.deleteMetadataAndAttributeSegments(storage, finalContainerId, timeout))
+                                        .thenAccept(z -> backUpMetadataSegments.put(finalContainerId, backUpMetadataSegment)),
+                                        ex -> Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException, null));
```

Comment:
returning `completableFuture` now. 
`synchroized` to avoid multiple threads from deleting and making copies.